src/UniGetUI/AutoUpdater.cs

```diff
@@ -262,11 +263,12 @@ private static async Task<bool> PrepairToLaunchInstaller(string installerLocatio
         }
         else
         {
-            Logger.Debug("Waiting for mainWindow to be closed or for user to trigger the update from the notification...");
+            Logger.Debug("Waiting for mainWindow to be closed, operations to complete, or for user to trigger the update from the notification...");
             while (
                 !ReleaseLockForAutoupdate_Window &&
                 !ReleaseLockForAutoupdate_Notification &&
-                !ReleaseLockForAutoupdate_UpdateBanner)
+                !ReleaseLockForAutoupdate_UpdateBanner &&
+                MainApp.Operations._operationList.Count == 0)
```

Comment:
Instead of usind the _operationList.Count (which also includes done operations, please use MainApp.Operations.AreThereRunningOperations())